File: commercial_evaluators/js/homedepot.js
var args = process.argv.slice(2);
var fs = require('fs');
var rs = fs.createReadStream(args[0]);
var ws = fs.createWriteStream(args[1]);
var rl = require('readline').createInterface(rs,ws);
rl.on('line', function (line) {
        ws.write(line+'\t'+tryPs(line)+'\n');
});
rl.on('close', function() {
          ws.end();
          process.exit(0);
});
function tryPs(pass) {
        /* adapted from homedepot script */
        switch (getPasswordLevel(pass)) {
          case 0:
            return "Weak";
            break;
          case 1:
            return "Good";
            break;
          case 2:
            return "Strong";
        }
      }
function getPasswordLevel(password) {
    var commonPasswordArray = ["12345678", "123123123", "123456789", "987654321", "1234567890", "1q2w3e4r", "1qaz2wsx", "abcd1234", "alexander", "asdfasdf", "asdfghjkl", "baseball", "chocolate", "computer", "football", "homedepot", "homedepot123", "iloveyou", "internet", "jennifer", "liverpool", "michelle", "password", "password1", "princess", "qwertyuiop", "sunshine", "superman", "testpassword", "trustno1", "welcome1", "whatever", "abcdefghi", "abcdefgh", "12345678"];
     //email = email !== undefined ? email : "";
	
	if ((password.length < 8) || (commonPasswordArray.indexOf(password.toLowerCase()) > -1) ||
        ((password.toLowerCase().split(password[0].toLowerCase()).length - 1) == password.length)) {
        return 0;
    }
    var score = 0;
    if (/[a-z]/.test(password)) { //Check lowercase
        score++;
    }
    if (/[A-Z]/.test(password)) { //Check uppercase
        score++;
    }
    if (/\d/.test(password)) { //Check Numbers
        score++;
    }
    if (/^[a-zA-Z0-9- ]*$/.test(password) == false) { //Check Special Characters
        score++;
    }
    if ((score > 1 && (password.length > 12)) || (score > 2 && (password.length > 8))) {
        return 2; //Strong Password
    }
    return 1; // Good Password
};

// $.inArray(password.toLowerCase(),commonPasswordArray)

File: homedepot.py
# Password strength evaluation 
# Written by Rutuparn Pawar (InputBlackBoxOutput)
# Created on 2 Oct 2020

from passwd import PasswordFeatures
import re

#-----------------------------------------------------------------------------------------------------
'''
Used by Home Depot (Retail Store)
Grading: 1-Weak 2-Good 3-Strong (Modified)
'''
class HomeDepot(PasswordFeatures):
	def __init__(self, password):
		super().__init__(password)
		self.commonPasswordArray = ["12345678", "123123123", "123456789", "987654321", "1234567890", "1q2w3e4r", "1qaz2wsx", "abcd1234", "alexander", "asdfasdf", "asdfghjkl", "baseball", "chocolate", "computer", "football", "homedepot", "homedepot123", "iloveyou", "internet", "jennifer", "liverpool", "michelle", "password", "password1", "princess", "qwertyuiop", "sunshine", "superman", "testpassword", "trustno1", "welcome1", "whatever", "abcdefghi", "abcdefgh", "12345678"];

	def get_score(self):
		if self.length < 8 or (self.password.lower() in self.commonPasswordArray) or ((len(self.password.lower().split(self.password[0].lower())) - 1) == self.length):
			return 1

		score = 0
		if self.lowercase_count():
		    score += 1

		if self.uppercase_count(): 
		    score += 1

		if self.digit_count():
		    score += 1

		if re.search("^[a-zA-Z0-9- ]*$", self.password):
			score += 1

		if (score > 1 and self.length > 12) or (score > 2 and self.length > 8):
		    return 3

		return 2

#-----------------------------------------------------------------------------------------------------
if __name__ == '__main__':
	homedepot = HomeDepot("CoolCat")
	print(homedepot.get_score())

#-----------------------------------------------------------------------------------------------------
# EOF